addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -3393,16 +3393,36 @@ X[]
                     const anchorOffset = selection.anchorOffset;
                     const p = editor.editable.querySelector('p');
                     const textNode = p.childNodes[0];
-                    triggerEvent(editor.editable, 'keydown', {key: ' ', code: 'Space'});
+                    triggerEvent(editor.editable, 'input', {data: ' ', inputType: 'insertText' });
                     textNode.textContent = "a http://test.com b http://test.com\u00a0 c http://test.com d";
                     selection.extend(textNode, anchorOffset + 1);
                     selection.collapseToEnd();
-                    triggerEvent(editor.editable, 'input', {data: ' ', inputType: 'insertText' });
+                    triggerEvent(editor.editable, 'keydown', {key: ' ', code: 'Space'});
                     triggerEvent(editor.editable, 'keyup', {key: ' ', code: 'Space'});
                 },
                 contentAfter: '<p>a http://test.com b <a href="http://test.com">http://test.com</a>&nbsp;[] c http://test.com d</p>',
             });
         });
+        it('should transform url after enter', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>a http://test.com b http://test.com[] c http://test.com d</p>',
+                stepFunction: async (editor) => {
+                    triggerEvent(editor.editable, 'keydown', {key: 'Enter'});
+                    triggerEvent(editor.editable, 'input', {data: ' ', inputType: 'insertParagraph' });
+                },
+                contentAfter: '<p>a http://test.com b <a href="http://test.com">http://test.com</a></p><p>[]&nbsp;c http://test.com d</p>',
+            });
+        });
+        it('should transform url after shift+enter', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>a http://test.com b http://test.com[] c http://test.com d</p>',
+                stepFunction: async (editor) => {
+                    editor.testMode = false;
+                    triggerEvent(editor.editable, 'keydown', {key: 'Enter', shiftKey: true});
+                },
+                contentAfter: '<p>a http://test.com b <a href="http://test.com">http://test.com</a><br>[]&nbsp;c http://test.com d</p>',
```

Comment:
Added this test and the one above (both pass) to illustrate that in those cases as well, the selection remains before the `&nbsp;`. Note as well the disappearance of the space before character `c` (seems unintended).

Note for later: remove `editor.testMode = false`.